### PR TITLE
Rhos stack cleanupcode

### PR DESCRIPTION
**Pull request by sshveta**
{{pytest: -v --long-running cfme/tests/services/test_provision_stack.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fb45ac063527a991be9f3111eb423d9c47b25fc5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: utils/mgmt_system/openstack.py [hunk 1]
```diff
@@ -621,3 +639,27 @@ def delete_template(self, template_name):
         template = self._find_template_by_name(template_name)
         template.delete()
         wait_for(lambda: not self.does_template_exist(template_name), num_sec=120, delay=10)
+
+    def stack_get(self, stack_id):
+        return self.stackapi.stacks.get(stack_id)
+
+    def stack_exist(self, stack_name):
+        stacks = [stack for stack in self.stack_get(stack_name)
+            if stack.stack_name == stack_name]
```
Comment (seandst) [inline utils/mgmt_system/openstack.py]: I mentioned some things along these lines in an email, but it looks like we need to be using stack_id at all times with the heat API. We've got a helpful function (maybe more than one) in the ec2 API that's very similar to this list comprehension; I think it makes sense to break this out into a function (like `_get_stack_id`, for example)

File: utils/mgmt_system/openstack.py [hunk 2]
```diff
@@ -621,3 +639,27 @@ def delete_template(self, template_name):
...
+        if (len(stacks) >= 1):
```
Comment (seandst) [inline utils/mgmt_system/openstack.py]: Pythonism: Any sequence with length >= 1 is considered "True", so there are a couple of ways to spell this conditional that are more "pythonic":

`return True if stacks else False`
```
if stacks:
  return True
else:
  return False
```

`return bool(stacks)`
(my favorite)

**Comment by dajoRH** [comment]
Lint report for commit e500c488f0af9d1b9b445da202261de4eee4f293:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ececcf378c6473b64f458951f94e1e5b8141479e:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5055e00626bf012dc56c5c62ec4a7a88a2d814e5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**